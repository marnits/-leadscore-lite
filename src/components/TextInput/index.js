import React from 'react';
import PropTypes from 'prop-types';
import Label from '../Label';
import styles from './styles.module.scss';

const TextInput = ({
  value,
  onChange,
  label,
  type,
  id,
}) => {
  const onInputChange = ({ target: { value: targetValue } }) => {
    onChange(targetValue);
  };

  return (
    <div className={styles.InputWithLabel}>
      {label && <Label htmlFor={id} text={label} />}
      <input
        type={type}
        id={id}
        onChange={onInputChange}
        className={`${styles.Input} ${value ? 'Input-initialized' : null}`}
        value={value}
      />
    </div>
  );
};

TextInput.defaultProps = {
  label: '',
  type: 'text',
};

TextInput.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
