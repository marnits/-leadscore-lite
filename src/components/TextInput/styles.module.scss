.Input {
  border-width: 0;
  border-radius: 6px;
  background-color: white;
  outline: none;
  padding: 6px 16px;
  font-size: 12px;
  background-color: #ffffff;
  color: #545454;
  font-weight: bold;
  width: 100%;
  box-sizing: border-box;
}

.Input:hover, .Input:focus {
  background-color: #f1efef;
}

.InputWithLabel {
  width: 100%;
  display: flex;
  justify-content: space-between;
}
