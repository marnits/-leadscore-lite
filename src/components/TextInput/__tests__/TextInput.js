import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TextInput from '../index';

describe('TextInput', () => {
  test('should render TextInput without label', () => {
    const tree = renderer.create(<TextInput id="input" value="" onChange={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render TextInput with label', () => {
    const tree = renderer.create(<TextInput label="test" id="input" value="" onChange={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should probably simulate changes inside input', () => {
    const props = {
      id: 'input',
      value: '',
      onChange: text => text,
    };

    const spy = jest.spyOn(props, 'onChange');
    const wrapper = shallow(<TextInput {...props} />);
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: 'changed' } });

    expect(spy).toHaveBeenCalledWith('changed');
  });
});
