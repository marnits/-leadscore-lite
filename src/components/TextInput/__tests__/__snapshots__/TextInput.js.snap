// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render TextInput 1`] = `
<div
  className="InputWithLabel"
>
  
  <input
    className="Input null"
    id="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput should render TextInput with label 1`] = `
<div
  className="InputWithLabel"
>
  <label
    className="Label"
    htmlFor="input"
  >
    test
  </label>
  <input
    className="Input"
    id="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput should render TextInput without label 1`] = `
<div
  className="InputWithLabel"
>
  <input
    className="Input"
    id="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
